src/game/emutest.c

```diff
@@ -0,0 +1,168 @@
+#include "emutest.h"
+
+#include <PR/os_internal_reg.h>
+#include <PR/os_internal_si.h>
+#include <PR/os_message.h>
+#include <PR/os_pi.h>
+#include <PR/R4300.h>
+#include <ultra64.h>
+#include <string.h>
+#include "types.h"
+
+extern OSMesgQueue gSIEventMesgQueue;
+extern u8 __osContPifRam[];
+extern u8 __osContLastCmd;
+extern void __osSiGetAccess(void);
+extern void __osSiRelAccess(void);
+
+enum Emulator gEmulator = EMU_CONSOLE;
+
+__attribute__((aligned(8)))
+static const u32 check_count_factor_asm[] = {
+    0x40084800u, // MFC0 T0, COUNT
+    0x40094800u, // MFC0 T1, COUNT
+    0x03E00008u, // JR RA
+    0x01281023u  // SUBU V0, T1, T0
+};
+
+static inline u32 check_count_factor() {
```

Comment:
This could be useful information too, care to throw the result of this into a global variable as well? Might also help to add comments that describe what the possible return values are.

Reply:
I guess I could. This will be the count factor on PJ64 1.6 and 3.0 and ParallelN64 (also Mupen64plus-next, but not if  Count Per Op Divider is set to anything other than 0 or 1). But I wouldn't really want to encourage its use outside of emulator detection. As you can see, it already doesn't work for PJ64 4.0